### ISSUE TITLE
PUPPETEER_SKIP_DOWNLOAD=true pnpm install was stuck long time

### ISSUE THREAD

**Issue by cxp-13** [elizaOS/eliza#1298]
I'm running the process of install and its stuck here. WSL2
```
cxp@R9000P:~/solana_learn/AI/eliza$ PUPPETEER_SKIP_DOWNLOAD=true pnpm install

Scope: all 46 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +4687
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 0, reused 3936, downloaded 0, added 4687, done
node_modules/onnxruntime-node: Running postinstall script...
node_modules/ffmpeg-static: Running install script, done in 13s
node_modules/@discordjs/opus: Running install script...
node_modules/canvas: Running install script, failed in 5.8s (skipped as optional)
node_modules/puppeteer: Running postinstall script, done in 141ms
node_modules/es5-ext: Running postinstall script, done in 47ms
node_modules/@discordjs/opus: Running install script, done in 35s
node_modules/@nomicfoundation/ethereumjs-tx/node_modules/secp256k1: Running install script, done in 69ms
node_modules/nx: Running postinstall script, done in 193ms
node_modules/@swc/core: Running postinstall script, done in 73ms
node_modules/wtf_wikipedia: Running postinstall script, done in 118ms
node_modules/@docusaurus/plugin-ideal-image/node_modules/sharp: Running install script, done in 8.8s
node_modules/unbuild/node_modules/esbuild: Running postinstall script, done in 77ms
node_modules/@multiversx/sdk-core/node_modules/keccak: Running install script, done in 69ms
node_modules/node-llama-cpp: Running postinstall script, done in 1s
```

**Comment by AIFlowML** [comment]
Hello here. 
I saw this same error in an old issue im runing al the backlog now. 
[https://github.com/elizaOS/eliza/issues/992](https://github.com/elizaOS/eliza/issues/992)
Let me know what version you run and maybe try latest one. 

Here some tips 
### TLDR (Executive Summary)
- **Issue Essence**: Long installation time for `PUPPETEER_SKIP_DOWNLOAD=true pnpm install` in WSL2.
- **Root Cause**: Large number of packages, optional dependencies, and WSL2 constraints.
- **Solution Approach**: Optimize dependencies, increase WSL2 resources, and consider alternative environments.
- **Impact Assessment**: Improved installation performance and reduced delays.

### Quick Resolution Path
1. **Check for Dependency Updates**:
   ```bash
   pnpm update
   ```
2. **Investigate Optional Dependencies**:
   ```bash
   pnpm list --depth=1
   ```
3. **Increase WSL2 Resources**:
   - Edit `.wslconfig` file:
     ```ini
     [wsl2]
     memory=4GB
     processors=2
     ```
4. **Use a Different Environment**:
   - Consider using a native Linux environment or Docker container.

### Comprehensive Solution
1. **Check for Dependency Updates**:
   - Ensure all dependencies are up to date to benefit from optimizations or fixes.
   ```bash
   pnpm update
   ```
2. **Investigate Optional Dependencies**:
   - Review optional dependencies to determine if they are critical and install them separately if needed.
   ```bash
   pnpm list --depth=1
   ```
3. **Increase WSL2 Resources**:
   - Allocate more resources to the WSL2 environment to improve installation performance.
   - Edit `.wslconfig` file:
     ```ini
     [wsl2]
     memory=4GB
     processors=2
     ```
4. **Use a Different Environment**:
   - Consider trying the installation in a native Linux environment or a different setup to compare performance.
5. **Optimize Installation**:
   - Use flags like `--prefer-offline` to reduce network calls and speed up the installation process.
   ```bash
   PUPPETEER_SKIP_DOWNLOAD=true pnpm install --prefer-offline
   ```
6. **Monitor Resource Usage**:
   - Use tools like `htop` or `top` to monitor CPU and memory usage during installation to identify bottlenecks.
   ```bash
   htop
   ```

**Comment by AIFlowML** [comment]
I cosider this an instalation related issue and not a bug.
Closing this now but need to check on the PUPPETEER_SKIP_DOWNLOAD flag for more robust handling.
